src/snowflake/cli/plugins/nativeapp/codegen/snowpark/extension_function_utils.py

```diff
@@ -13,6 +13,107 @@
 TEMP_OBJECT_NAME_PREFIX = "SNOWPARK_TEMP_"
 
 
+def add_defaults_to_extension_function(ex_fn: Dict[str, Any]) -> bool:
+    """
+    Helper function to add in defaults for keys that do not exist in the dictionary or contain empty strings when they should not.
+    This helper function is needed because different callback sources can create dictionaries with different/missing keys, and since
+    Snowflake CLI may not own all callback implementations, the dictionaries need to have the minimum set of keys and their default
+    values to be used in creation of the SQL DDL statements.
+
+    Returns:
+    A boolean value, True if everything has been successfully validated and assigned, False if an error was encountered.
+    """
+
+    # Must have keys
+    try:
+        ex_fn["object_type"] = ex_fn["object_type"].upper()
+        assert ex_fn["object_type"] is not ""
+        ex_fn["object_name"] = ex_fn["object_name"].upper()
+        assert ex_fn["object_name"] is not ""
+        ex_fn["return_sql"] = ex_fn["return_sql"].upper()
+        assert ex_fn["return_sql"] is not ""
+    except KeyError as err:
+        cc.warning(f"{err}")
+        return False
+    except AssertionError as err:
+        cc.warning(f"{err}")
+        return False
+
+    # Other optional keys
+    ex_fn["anonymous"] = ex_fn.get("anonymous", False)
+    ex_fn["replace"] = ex_fn.get("replace", False)
+    ex_fn["if_not_exists"] = ex_fn.get("if_not_exists", False)
+
+    if ex_fn["replace"] and ex_fn["if_not_exists"]:
+        cc.warning("Options 'replace' and 'if_not_exists' are incompatible.")
+        return False
+
+    try:
+        assert len(ex_fn["input_args"]) == len(ex_fn["input_sql_types"])
+    except AssertionError as err:
+        cc.warning(
+            "The number of function parameters does not match the number of parameter types."
+        )
+        return False
+
+    has_imports = ex_fn.get("all_imports", None) and len(ex_fn["all_imports"]) > 0
+    ex_fn["all_imports"] = ex_fn["all_imports"] if has_imports else None
+
+    has_packages = ex_fn.get("all_packages", None) and len(ex_fn["all_packages"]) > 0
+    ex_fn["all_packages"] = ex_fn["all_packages"] if has_packages else None
+
+    has_eai = (
+        ex_fn.get("external_access_integrations", None)
+        and len(ex_fn["external_access_integrations"]) > 0
+    )
+    ex_fn["external_access_integrations"] = (
+        ex_fn["external_access_integrations"] if has_eai else None
+    )
+
+    has_secrets = ex_fn.get("secrets", None) and len(ex_fn["secrets"]) > 0
+    ex_fn["secrets"] = ex_fn["secrets"] if has_secrets else None
```

Comment:
that's a lot of code to sanitize fields that are optional. Not sure if we really need it, something like:
```
secrets = ex_fn.get("secrets")
if secrets:
     pass  # use it
```

doesn't seem so bad. If we do want this much sanitization, we could just introduce a helper to do it for multiple fields:

```
sanitize_list_attribute(ex_fn, "secrets")
sanitize_list_attribute(ex_fn, "inline_python_code")
...
```